sdk/ml/azure-ai-ml/azure/ai/ml/entities/_monitoring/definition.py

```diff
@@ -83,18 +89,41 @@ def _to_rest_object(self) -> RestMonitorDefinition:
         )
 
     @classmethod
-    def _from_rest_object(cls, obj: RestMonitorDefinition) -> "MonitorDefinition":
+    def _from_rest_object(cls, obj: RestMonitorDefinition, **kwargs) -> "MonitorDefinition":
+        tags = kwargs.get("tags")
         from_rest_alert_notification = None
         if obj.alert_notification_setting:
             if isinstance(obj.alert_notification_setting, AzMonMonitoringAlertNotificationSettings):
                 from_rest_alert_notification = AZMONITORING
             else:
                 from_rest_alert_notification = AlertNotification._from_rest_object(obj.alert_notification_setting)
         return cls(
-            compute=obj.compute_id,
+            compute=SparkResourceConfiguration(
+                instance_type=tags.pop(SPARK_INSTANCE_TYPE_KEY),
+                runtime_version=tags.pop(SPARK_RUNTIME_VERSION),
+            ),
             monitoring_target=MonitoringTarget(endpoint_deployment_id=obj.monitoring_target),
             monitoring_signals={
                 signal_name: MonitoringSignal._from_rest_object(signal) for signal_name, signal in obj.signals.items()
             },
             alert_notification=from_rest_alert_notification,
         )
+
+    def _populate_default_signal_information(
+        self,
+        model_inputs_arm_id: str,
+        model_inputs_type: str,
+        model_outputs_arm_id: str,
+        model_outputs_type: str,
+    ):
+        self.monitoring_signals = {
+            DefaultMonitorSignalNames.DATA_DRIFT_SIGNAL: DataDriftSignal._get_default_data_drift_signal(
```

Comment:
just to make sure, this creates the default collection but only gets used if the fields aren't set right?

Reply:
yes, this only gets created if a deployment id is given and there are no signals